runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/BatchStatefulParDoOverrides.java

```diff
@@ -176,7 +176,7 @@ private MultiOutputOverrideFactory(boolean isFnApi) {
     public PCollection<OutputT> expand(PCollection<KV<K, InputT>> input) {
       DoFn<KV<K, InputT>, OutputT> fn = originalParDo.getFn();
       verifyFnIsStateful(fn);
-      DataflowRunner.verifyStateSupported(fn);
+      DataflowRunner.verifyDoFnSupported(fn, false);
```

Comment:
Passing a raw bool into a function call is not very readable. I suggest splitting into `verifyDoFnSupportedForStreaming` and `verifyDoFnSupportedForBatch`. These can each call the common code.

Reply:
:+1:

Reply:
I added both versions, although the second one `verifyDoFnSupportedStreaming` is not used. I added that so that the methods are not imbalanced. The streaming case is called from `DataflowRunner.verifyDoFnSupported(fn, context.getPipelineOptions().isStreaming())`, where it would be weird to do `if (context..isStream()) verifyStreaming() else ...`